utils/usage/visibility_system_manager.go

```diff
@@ -0,0 +1,68 @@
+package usage
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+)
+
+type VisibilitySystemManager struct {
+	serverDetails *config.ServerDetails
+}
+
+func NewVisibilitySystemManager(serverDetails *config.ServerDetails) *VisibilitySystemManager {
+	return &VisibilitySystemManager{
+		serverDetails: serverDetails,
+	}
+}
+
+type labels struct {
+	ProductID                            string `json:"product_id"`
+	FeatureID                            string `json:"feature_id"`
+	OIDCUsed                             string `json:"oidc_used"`
+	JobID                                string `json:"job_id"`
+	RunID                                string `json:"run_id"`
+	GitRepo                              string `json:"git_repo"`
+	GhTokenForCodeScanningAlertsProvided string `json:"gh_token_for_code_scanning_alerts_provided"`
+}
+
+type visibilityMetric struct {
+	Value       int    `json:"value"`
+	MetricsName string `json:"metrics_name"`
+	Labels      labels `json:"labels"`
+}
+
+func (vsm *VisibilitySystemManager) createMetric(commandName string) ([]byte, error) {
```

Comment:
why dont we use VisibilityMetric as a return type, to enforce this type on the manager.LogMetric() func?

Reply:
That indeed makes sense @sverdlov93. Thanks!
Let me implement this in a follow-up PR, after this upcoming initial release.